### PR TITLE
feat(credential): Use credential type when issuing credentials

### PR DESCRIPTION
**Pull request by mgaffney**
This change uses the credential type attribute when issuing credentials
including mapping overrides.

### CODE REVIEW

File: internal/credential/vault/private_library.go [hunk 1]
```diff
@@ -5,91 +5,146 @@ import (
...
+	return bc, nil
+}
+
+func baseToUsrPass(ctx context.Context, bc *baseCred) (*usrPassCred, error) {
+	uAttr, pAttr := bc.lib.UsernameAttribute, bc.lib.PasswordAttribute
+	if uAttr == "" {
+		uAttr = "username"
+	}
+	if pAttr == "" {
+		uAttr = "password"
```
Comment (louisruch) [inline internal/credential/vault/private_library.go:61]:
```suggestion
		pAttr = "password"
```

**Review by louisruch — APPROVED** [review]
Took a quick look, will review properly tomorrow.  But found one bug

File: internal/credential/vault/private_library.go [hunk 2]
```diff
@@ -5,91 +5,146 @@ import (
...
+	}
+
+	var username, password string
+	if u, ok := bc.secretData[uAttr]; ok {
+		username = fmt.Sprintf("%v", u)
+	}
+	if p, ok := bc.secretData[pAttr]; !ok {
```
Comment (louisruch) [inline internal/credential/vault/private_library.go:68]: Haha, so I was trying to figure out why the test was passing given the not setting `pAttr` above, the `!ok` should become `ok`.

What this was doing was setting `username` = passwordAttribue and `password` = `<nil>` :)

Comment (mgaffney) [reply internal/credential/vault/private_library.go]: I fixed it and added a bunch of tests.

**Review by tmessi — APPROVED** [review]

**Pull request merged**